dll/shellext/zipfldr/CZipExtract.cpp

```diff
@@ -390,9 +399,16 @@ class CZipExtract :
             CStringA FullPath = CombinedPath;
             FullPath.Replace('/', '\\');    /* SHPathPrepareForWriteA does not handle '/' */
             DWORD dwFlags = SHPPFW_DIRCREATE | (is_dir ? SHPPFW_NONE : SHPPFW_IGNOREFILENAME);
+        RetryPrepare:
             HRESULT hr = SHPathPrepareForWriteA(hDlg, NULL, FullPath, dwFlags);
             if (FAILED_UNEXPECTEDLY(hr))
             {
+                char StrippedPath[MAX_PATH] = { 0 };
+
+                StrCpyNA(StrippedPath, BaseDirectory, _countof(StrippedPath));
+                PathStripPathA(StrippedPath);
+                if (WriteErrorPrompt(hDlg, (LPCSTR)&StrippedPath, hr, FALSE) == IDRETRY)
+                    goto RetryPrepare;
```

Comment:
tbh I don't really like the use of goto here, since this is already a fairly complex function with a double loop and multiple different branches.

Reply:
Well, both goto's here are merely rollbacks to previous position in code, and they both act within a single while loop without breaking its bounds (yes, there are other loops, but they're untouched).

Reply:
Ok you convinced me to remove this one :) It's now only warning box displayed here.